src/checkers/chromiumchecker.py

```diff
@@ -0,0 +1,183 @@
+    ExternalDataSource,
+    ExternalGitRepo,
+    ExternalGitRef,
+    ExternalGitRepoSource,
+)
+from ..lib.utils import get_extra_data_info_from_url
+
+log = logging.getLogger(__name__)
+
+
+class Component:
+    NAME: str
+    DATA_CLASS: t.Type[ExternalBase]
+
+    def __init__(
+        self,
+        session: aiohttp.ClientSession,
+        external_data: ExternalBase,
+        latest_version: str,
+    ) -> None:
+        self.session = session
+        self.external_data = external_data
+        self.latest_version = latest_version
+
+        assert latest_version is not None
+
+    async def check(self) -> None:
+        raise NotImplemented
+
+    async def update_external_source_version(self, latest_url):
+        assert latest_url is not None
+
+        try:
+            new_version = await get_extra_data_info_from_url(latest_url, self.session)
+        except (
+            aiohttp.ClientError,
+            aiohttp.ServerConnectionError,
+            aiohttp.ServerDisconnectedError,
+            aiohttp.ServerTimeoutError,
+        ) as e:
+            log.warning("%s returned %s", latest_url, e)
+            self.external_data.state = ExternalData.State.BROKEN
+        else:
+            new_version = new_version._replace(  # pylint: disable=no-member
+                version=self.latest_version
+            )
+            self.external_data.set_new_version(new_version)
+
+
+class ChromiumComponent(Component):
+    NAME = "chromium"
+    DATA_CLASS = ExternalDataSource
+
+    _URL_FORMAT = (
+        "https://commondatastorage.googleapis.com"
+        "/chromium-browser-official/chromium-{version}.tar.xz"
+    )
+
+    async def check(self) -> None:
+        assert isinstance(self.external_data, ExternalDataSource)
+
+        latest_url = self._URL_FORMAT.format(version=self.latest_version)
+        await self.update_external_source_version(latest_url)
+
+
+class LLVMVersionUtilsMixin:
+    class Version(t.NamedTuple):
+        revision: str
+        sub_revision: str
+
+    _UPDATE_PY_URL_FORMAT = (
+        "https://chromium.googlesource.com/chromium/src/+"
+        "/{version}/tools/clang/scripts/update.py"
+    )
+
+    _UPDATE_PY_PARAMS = {"format": "TEXT"}
+
+    _CLANG_REVISION_RE = re.compile(r"CLANG_REVISION = '(.*)'")
+    _CLANG_SUB_REVISION_RE = re.compile(r"CLANG_SUB_REVISION = (\d+)")
+
+    async def get_llvm_version(self) -> "LLVMVersionUtilsMixin.Version":
+        url = self._UPDATE_PY_URL_FORMAT.format(version=self.latest_version)
+        async with self.session.get(url, params=self._UPDATE_PY_PARAMS) as response:
+            result = await response.text()
+
+        update_py = base64.b64decode(result).decode("utf-8")
+        revision = self._CLANG_REVISION_RE.search(update_py).group(1)
+        sub_revision = self._CLANG_SUB_REVISION_RE.search(update_py).group(1)
+
+        return LLVMVersionUtilsMixin.Version(revision, sub_revision)
+
+
+class LLVMGitComponent(Component, LLVMVersionUtilsMixin):
```

Comment:
Maybe use one more level of inheritance instead of the mixin, e.g. `Component -> LLVMComponent -> LLVMGitComponent`? This will also make Mypy happier.